### PR TITLE
Customize env variable

### PR DESCRIPTION
**Pull request by helenyuyu**
I bumped source compatibility to 1.8 because I wanted to use lambdas, but I'm happy to revert this if there's a reason that we use 1.7.

### CODE REVIEW

File: src/main/java/com/palantir/gradle/circlestyle/CircleStylePlugin.java [hunk 1]
```diff
@@ -15,57 +15,49 @@
...
-                });
-                project.getTasks().withType(FindBugs.class, new Action<FindBugs>() {
-                    @Override
-                    public void execute(FindBugs findbugsTask) {
-                        configureFindbugsTask(project, findbugsTask, circleReportsDir, timer);
-                    }
-                });
+        CircleStylePluginExtension extension = rootProject.getExtensions()
+                .create("circleStyle", CircleStylePluginExtension.class);
+        rootProject.afterEvaluate(project -> {
```
Comment (jiahuijiang) [inline src/main/java/com/palantir/gradle/circlestyle/CircleStylePlugin.java]: project -> evaluatedRootProject; project1 -> project

File: src/test/resources/com/palantir/gradle/circlestyle/build.gradle_envVariableOverride [hunk 2]
```diff
@@ -0,0 +1,50 @@
+plugins {
+  id 'java'
+  id 'checkstyle'
+  id 'findbugs'
+}
+
```
Comment (jiahuijiang) [inline src/test/resources/com/palantir/gradle/circlestyle/build.gradle_envVariableOverride:6]: empty line --

File: src/test/java/com/palantir/gradle/circlestyle/CircleStylePluginTests.java [hunk 3]
```diff
@@ -35,8 +25,17 @@
...
-import com.google.common.base.Joiner;
-import com.google.common.io.Files;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Charsets.UTF_8;
```
Comment (jiahuijiang) [inline src/test/java/com/palantir/gradle/circlestyle/CircleStylePluginTests.java:35]: move static import back to the top of the file

File: src/test/java/com/palantir/gradle/circlestyle/CircleStylePluginTests.java [hunk 4]
```diff
@@ -73,6 +72,26 @@ public void checkstyleIntegrationTest() throws IOException {
         assertThat(reportXml).contains("Name 'a_constant' must match pattern");
     }
 
+    @Test
+    public void checkstyleIntegrationTestWithOverridenEnvVariableName() throws IOException {
+        copyTestFile("build.gradle_envVariableOverride", projectDir, "build.gradle");
+        env.set("CIRCLE_TEST_REPORTS", null);
```
Comment (jiahuijiang) [inline src/test/java/com/palantir/gradle/circlestyle/CircleStylePluginTests.java:78]: instead of unset this, set it to some random different directory? :D

**Review by jiahuijiang — APPROVED** [review]

**Comment by alicederyn** [comment]
I'd prefer to leave the source compatibility as it was

File: src/test/java/com/palantir/gradle/circlestyle/CircleStylePluginTests.java (hunk 3 above)
Comment (helenyuyu) [reply src/test/java/com/palantir/gradle/circlestyle/CircleStylePluginTests.java:35]: hmm my IDE hates this and keeps moving it back down here

**Comment by helenyuyu** [comment]
@alicederyn reverted the source compatibility changes

**Comment by helenyuyu** [comment]
@alicederyn would you be able to take a look at this pr when you have the time? we're somewhat blocked on this feature

File: src/main/java/com/palantir/gradle/circlestyle/CircleStylePlugin.java [hunk 5]
```diff
@@ -15,46 +15,51 @@
...
 package com.palantir.gradle.circlestyle;
 
-import java.io.File;
-
-import org.gradle.api.Action;
-import org.gradle.api.Plugin;
-import org.gradle.api.Project;
-import org.gradle.api.Task;
-import org.gradle.api.UnknownTaskException;
+import org.gradle.api.*;
```
Comment (alicederyn) [inline src/main/java/com/palantir/gradle/circlestyle/CircleStylePlugin.java:18]: Not ideal

**Review by alicederyn — APPROVED** [review]
Sorry for the delay. Approving and merging as the only objections are about imports, but if you could restore the ordering and remove the * import in another PR I'd be grateful.

**Pull request merged**